### PR TITLE
vdiff: factor Summarize out from vtctldclient

### LINKED ISSUE #16780: Feature Request: export vtctldclient VDiffShow summarization code
### Feature Description

When executing `vtctldclient vdiff show`, a summary is generated based on the raw RPC outputs containing query rows, fields, etc. This summary is useful for humans using the CLI, but is also useful for automating the VDiff flow. Being able to import the summary logic as a library means that callers don't need to parse the entire `VDiffShowResponse` type to generate a report similar to `vtctldclient`.

### Use Case(s)

Creating workflow automation tooling in Go, and being able to programmatically check the validity of workflows via VDiff.

### PR DESCRIPTION
**Pull request by mdlayher**
## Description

Factor out buildSingleSummary as vdiff.Summarize, which generates a summary of the VDiff operation based on the VDiffShow RPC response.

## Related Issue(s)

  - Fixes: #16780

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported to release branches
-   [x] If this change is to be back-ported to previous releases, a justification is included in the PR description
-   [x] Tests were added or are not required
-   [x] Did the new or modified tests pass consistently locally and on CI?
-   [x] Documentation was added or is not required

### CODE REVIEW

File: go/cmd/vtctldclient/command/vreplication/vdiff/vdiff.go [hunk 1]
```diff
@@ -438,7 +408,7 @@ HasMismatch:  {{.HasMismatch}}
 StartedAt:    {{.StartedAt}}
 {{if (eq .State "started")}}Progress:     {{printf "%.2f" .Progress.Percentage}}%%{{if .Progress.ETA}}, ETA: {{.Progress.ETA}}{{end}}{{end}}
 {{if .CompletedAt}}CompletedAt:  {{.CompletedAt}}{{end}}
-{{range $table := .TableSummaryMap}} 
+{{range $table := .TableSummaryMap}}
```
Comment (mdlayher) [inline go/cmd/vtctldclient/command/vreplication/vdiff/vdiff.go:411]: Small bit of trailing whitespace here and on line 420. I suspect it's not load bearing but if it turns out to be, I can undo this.

**Comment by mattlord** [comment]
@mdlayher I like this idea! Makes total sense to me. I don't, however, think that `tabletmanager` is the right place for this code as it's really `VtctldServer` code that's working with `vtctldata` protos (rather than `TabletManager` code working with `tabletmanagerdata` protos), which primarily lives in the `workflow` package (`go/vt/vtctl/workflow`). I think we should probably create a `go/vt/vtctl/workflow/vdiff.go` file and move this code there along with the other VDiff functions that are currently in `workflow/server.go` — we need to start splitting that file up anyway as it's getting massive. That make sense?

**Comment by deepthi** [comment]
Closing as stale. Please re-open if you would like to address review feedback.

**Comment by mdlayher** [comment]
Thanks! I owe a new PR with a different approach.

**Pull request closed without merging**